src/persist/src/operators/stream.rs

```diff
@@ -767,9 +769,55 @@ mod tests {
         let (_write, primary_read) = p.create_or_load::<(), ()>("primary");
         let (_write, condition_read) = p.create_or_load::<(), ()>("condition");
 
-        let (primary_listen_rx, _) = primary_read.listen()?;
-        let (condition_listen_rx, _) = condition_read.listen()?;
+        #[derive(Copy, Clone, Debug, PartialEq, Eq)]
+        enum Sealed {
+            Primary(u64),
+            Condition(u64),
+        }
+
+        let (primary_listen_tx, primary_listen_rx) = crossbeam_channel::unbounded();
+        primary_read.listen(primary_listen_tx)?;
+
+        let (condition_listen_tx, condition_listen_rx) = crossbeam_channel::unbounded();
+        condition_read.listen(condition_listen_tx)?;
+        let (listen_tx, listen_rx) = crossbeam_channel::unbounded();
+
+        let runtime = Runtime::new()?;
+        let listener_handle = runtime.spawn(async move {
+            let mut num_finished = 0;
```

Comment:
took me a bit to figure out what this was doing. I think the confusion can be avoid with a slightly more specific var name for this. how about "num_channels_closed" or something?